Added the Dune 2000 Guarding voice

Mailaender opened the pull request:
as requested in https://github.com/OpenRA/OpenRA/pull/7147.

Mailaender committed:
setup the Guarding voice

Phrohdoh commented:
The *Info reference should be readonly.

Mailaender commented:
Right. Updated.

Phrohdoh commented:
Looks good to me but I can't test atm.

ScottNZ commented on OpenRA.Mods.RA/Guard.cs:
```diff
@@ -18,10 +18,22 @@
-	class GuardInfo : TraitInfo<Guard> { }
+	class GuardInfo : ITraitInfo
+	{
+		public readonly string VoicePhrase = "Move";
```
This field should be simply called `Voice`. See C4Demolition.cs, Selectable.cs, etc

Mailaender replied:
Sure thing. Updated.

obrakmann commented:
:+1:

penev92 commented:
TD, RA and TS show no regression.
Sounds good in D2k, except for the sardaukar, which currently crash the game with
>System.InvalidOperationException was unhandled
  Message=Can't find Guard in voice pool.
  StackTrace:
       at OpenRA.Sound.PlayPredefined(Ruleset ruleset, Player p, Actor voicedUnit, String type, String definition, String variant, Boolean relative, WPos pos, Single volumeModifier, Boolean attenuateVolume) in OpenRA.Game\Sound\Sound.cs:line 331

Also why are the the Fremen and the Saboteurs using the Atreides and Ordos sounds respectively, regardless of what house you are?

penev92 commented on OpenRA.Mods.RA/Guard.cs:
```diff
@@ -18,10 +18,22 @@
+		public object Create(ActorInitializer init) { return new Guard(this); }
+	}
 
 	class Guard : IResolveOrder, IOrderVoice
```
Can we rename this to `Guards` ?
`Guard` sounds more like like an activity than a trait.

Mailaender replied:
Definitely not in this pull request. Make a separate ticket for that.

penev92 replied:
OK.

Mailaender commented:
Rebased and fixed the crash regarding Sardaukar.

> Also why are the Fremen and the Saboteurs using the Atreides and Ordos sounds respectively, regardless of what house you are?

Probably because they are special units and they did not record for houses that usually can't train them.

penev92 commented:
But the "Guarding" notification is not unit-specific. The ... "House assistant", let's call it, says it. (You know, the one that says "Building" and stuff).

penev92 commented:
OK I don't know how it was in the original game, but if we chose not to use player-House's-voice for these House-specific units, the Sardaukar will need to use the Harkonnen voice.

Unit158 commented:
Sardaukar had their own sound in the original.

On 2015-01-06, at 3:33 PM, penev92 <notifications@github.com> wrote:

> OK I don't know how it was in the original game, but if we chose not to use player-House's-voice for these House-specific units, the Sardaukar will need to use the Harkonnen voice.
> 
> \
> Reply to this email directly or view it on GitHub.
>

huwpascoe commented:
Can confirm this works :+1:

Mailaender commented:
Don't merge this yet. I can make the guarding voices for the special units current house specific, too.

penev92 commented:
I think that would feel less strange.

penev92 commented:
> Can confirm this works :+1:

Partially.

Mailaender commented:
Updated.

penev92 commented:
This now works :+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/c3a43712de1a590f53977d7e668de194ebc1d5b9%5E...c3a43712de1a590f53977d7e668de194ebc1d5b9)
